src/main/kotlin/com/zepben/evolve/services/network/translator/NetworkProtoToCim.kt

```diff
@@ -541,6 +548,14 @@ fun toCim(pb: PBConnectivityNode, networkService: NetworkService): ConnectivityN
 fun toCim(pb: PBConnectivityNodeContainer, cim: ConnectivityNodeContainer, networkService: NetworkService): ConnectivityNodeContainer =
     cim.apply { toCim(pb.psr, this, networkService) }
 
+fun toCim(pb: PBCurve, cim: Curve, networkService: NetworkService): Curve =
+    cim.apply {
+        pb.curveDataList.forEach {
+            addCurveData(CurveData(it.xvalue, it.y1Value, it.y2Value, it.y3Value))
```

Comment:
todo: should be using `UNKNOWN_DOUBLE` check. Need to check why testing has not detected this.

Reply:
added the check

Reply:
Did you find out why this wasn't detected in the tests? I assume this is because you haven't added the classes to `NetworkTranslatorTest`

Reply:
Added to test and the missing toPB functions.

Reply:
Changed up SynchronousMachine's earthing to non-nullable boolean and table/table creation.